pkg/client/example-runner.go

```diff
@@ -0,0 +1,287 @@
+package client
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"github.com/GoogleCloudPlatform/gcp-service-broker/pkg/broker"
+	"github.com/pivotal-cf/brokerapi"
+)
+
+func RunExamples(client *Client) error {
+	return RunExamplesForService(client, "")
+}
+
+func RunExamplesForService(client *Client, serviceName string) error {
+	rand.Seed(time.Now().UTC().UnixNano())
+
+	services := broker.GetAllServices()
+
+	for _, service := range services {
+
+		if serviceName != "" && serviceName != service.Name {
+			continue
+		}
+
+		for _, example := range service.Examples {
+			err := RunExample(client, example, service)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+
+}
+
+func RunExample(client *Client, example broker.ServiceExample, service *broker.BrokerService) error {
+	executor, err := newExampleExecutor(client, example, service)
+	if err != nil {
+		return err
+	}
+
+	executor.LogTestInfo()
+
+	// Cleanup the test if it fails partway through
+	defer func() {
+		log.Println("Cleaning up the environment")
+		executor.Unbind()
+		executor.Deprovision()
+	}()
+
+	if err := executor.Provision(); err != nil {
+		return err
+	}
+
+	bindResponse, err := executor.Bind()
+	if err != nil {
+		return err
+	}
+
+	if err := executor.Unbind(); err != nil {
+		return err
+	}
+
+	if err := executor.Deprovision(); err != nil {
+		return err
+	}
+
+	// Check that the binding response has the same fields as expected
+	var binding brokerapi.Binding
+	err = json.Unmarshal(bindResponse, &binding)
+	if err != nil {
+		return err
+	}
+
+	credentialsEntry := binding.Credentials.(map[string]interface{})
```

Comment:
I understand that this isn't our API, but man is this brittle. </rant>

Reply:
Thanks, this reminded me to open #226!